libkbfs/block_retrieval_worker.go

```diff
@@ -57,7 +57,12 @@ func (brw *blockRetrievalWorker) HandleRequest() (err error) {
 	}
 
 	// Create a new block of the same type as the first request
-	block := retrieval.requests[0].block.NewEmpty()
+	var block Block
+	retrieval.reqMtx.RLock()
+	if retrieval.requests[0].block != nil {
```

Comment:
What happens when it is `nil`?  I don't think we can safely pass a typeless, `nil` block into `getBlock` below, can we?